core/src/test/java/org/infinispan/container/offheap/OffHeapConcurrentMapTest.java

```diff
@@ -0,0 +1,131 @@
+      offHeapEntryFactory.start();
+
+      map = new OffHeapConcurrentMap(allocator, offHeapEntryFactory, null);
+      map.start();
+   }
+   @AfterMethod
+   void afterMethod() {
+      if (map != null) {
+         map.stop();
+      }
+   }
+
+   public void testIterationStartedWithResize1() {
+      int resizeLimitation = OffHeapConcurrentMap.INITIAL_SIZE - (OffHeapConcurrentMap.INITIAL_SIZE >> 2);
+
+      WrappedByteArray valueByteArray = new WrappedByteArray(new byte[] { 0, 1, 2, 3, 4, 5 });
+      for (int i = 0; i < resizeLimitation - 1; ++i) {
+         putInMap(map, valueByteArray);
+      }
+
+      Iterator<InternalCacheEntry<WrappedBytes, WrappedBytes>> iterator = map.values().iterator();
+      AssertJUnit.assertTrue(iterator.hasNext());
+      Assert.assertNotNull(iterator.next());
+      int entriesFound = 1;
+
+      putInMap(map, valueByteArray);
+
+      while (iterator.hasNext()) {
+         Assert.assertNotNull(iterator.next());
+         entriesFound++;
+      }
+
+      // Now we have to make sure that our entry count matches or is 1 greater than the amount to resize
+      // The last one can vary if it was in a bucket before or after our iteration current position
+      if (entriesFound != resizeLimitation && entriesFound != resizeLimitation - 1) {
+         AssertJUnit.fail("Iterator returned incorrect number expect " + resizeLimitation + " or " + (resizeLimitation - 1) +
+               ", but got " + entriesFound);
+      }
+   }
+
+   public void testIterationStartedWithTwoResizes() {
+      int resizeLimitation = OffHeapConcurrentMap.INITIAL_SIZE - (OffHeapConcurrentMap.INITIAL_SIZE >> 2);
+
+      WrappedByteArray valueByteArray = new WrappedByteArray(new byte[] { 0, 1, 2, 3, 4, 5 });
+      for (int i = 0; i < resizeLimitation - 1; ++i) {
+         putInMap(map, valueByteArray);
+      }
+
+      Iterator<InternalCacheEntry<WrappedBytes, WrappedBytes>> iterator = map.values().iterator();
+      AssertJUnit.assertTrue(iterator.hasNext());
+      Assert.assertNotNull(iterator.next());
+      int entriesFound = 1;
+
+      // Now cause a first and second
+      for (int i = 0; i < resizeLimitation + 2; ++i) {
+         putInMap(map, valueByteArray);
+      }
+
+      while (iterator.hasNext()) {
+         Assert.assertNotNull(iterator.next());
+         entriesFound++;
+      }
+
+      int expect = (int) ((resizeLimitation * 2) * .95);
+      // Unfortunately we don't know exactly how many entries it had to return due to not knowing the starting bucket
+      // during iteration or how many entries were inserted before or after - however it is very unlikely that we lost
+      // 5% of our entries
+      if (entriesFound < expect) {
+         AssertJUnit.fail("The amount of entries found was: " + entriesFound + ", but we should probably have more than " + expect);
+      }
+   }
+
+   public void testIterationCreatedButNotStartedWithResize() {
```

Comment:
I can add that test, the reason I didn't was because it doesn't really test much, since the iteration is done lazily, so you actually have to start iterating over it and have a resize for anything special to happen.